tidb.go

```diff
@@ -111,6 +112,24 @@ func Parse(ctx context.Context, src string) ([]ast.StmtNode, error) {
 	return stmts, nil
 }
 
+// Before every execution, we must clear statement status.
+func resetStmtStatus(ctx context.Context, s ast.StmtNode) {
+	sessVars := ctx.GetSessionVars()
+	var sc variable.StatementContext
+	switch stmt := s.(type) {
+	case *ast.ShowStmt:
+		if stmt.Tp == ast.ShowWarnings {
+			sc.SetWarnings(sessVars.StmtCtx.GetWarnings())
```

Comment:
No, session variable is created by `NewSessionVars`, in which `StmtCtx` is always created.